### PR TITLE
Add git to command palette

### PR DESCRIPTION
**Pull request by tsabbir96**
An attempt to develop the feature requested at #1238

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -137,6 +139,22 @@ async function activate(
     );
     return null;
   }
+  // Add commands to the pallette
+  const command = CommandIDs.gitClone;
+
+  commands.addCommand(command, {
```
[the remark this answers is not in the document]
Comment (tsabbir96) [reply src/index.ts:145]: Understood, thanks for explaining!

**Comment by tsabbir96** [comment]
@fcollonval Hi, I adjusted the changes that you requested. Looks like the PR is failing an Integration Test (UI Test). Here's the generated log:

```
Running 18 tests using 1 worker
··············F···

  1) tests/merge-commit.spec.ts:59:7 › Merge commit tests › should diff file after clicking ────────

    Error: proxy.waitForSelector: Target closed
    =========================== logs ===========================
    waiting for locator('.jp-git-diff-root') to be visible
    ============================================================

      69 |       .waitFor({ state: 'visible' });
      70 |
    > 71 |     expect(page.waitForSelector('.jp-git-diff-root')).toBeTruthy();
         |                 ^
      72 |   });
      73 |
      74 |   test('should revert merge commit', async ({ page }) => {

        at /home/runner/work/jupyterlab-git/jupyterlab-git/ui-tests/tests/merge-commit.spec.ts:71:[17](https://github.com/jupyterlab/jupyterlab-git/actions/runs/4788641195/jobs/8515773749?pr=1228#step:12:18)

    Test finished within timeout of 60000ms, but tearing down "context" ran out of time.
    Please allow more time for the test, since teardown is attributed towards the test timeout budget.

    attachment #1: video (video/webm) ──────────────────────────────────────────────────────────────
    test-results/tests-merge-commit-Merge-commit-tests-should-diff-file-after-clicking/video.webm
    ────────────────────────────────────────────────────────────────────────────────────────────────

  1 failed
    tests/merge-commit.spec.ts:59:7 › Merge commit tests › should diff file after clicking ─────────
  17 passed (1.8m)
error Command failed with exit code 1.
```

Any thought on this?

**Comment by tsabbir96** [comment]
Hi @fcollonval sorry to bother you again. Could you provide some insights on this? I am stuck here and any feedback will be much appreciated.

File: src/index.ts [hunk 2]
```diff
@@ -185,6 +193,23 @@ async function activate(
     gitPlugin.title.icon = gitIcon;
     gitPlugin.title.caption = 'Git';
 
+    // Add the commands to the command palette
+    const category = 'Git Operations';
```
Comment (fcollonval) [inline src/index.ts:197]: let's internationalize that string:

```suggestion
    const category = trans.__('Git Operations');
```

File: src/index.ts [hunk 3]
```diff
@@ -185,6 +193,23 @@ async function activate(
...
+    [
+      CommandIDs.gitToggleSimpleStaging,
+      CommandIDs.gitToggleDoubleClickDiff,
+      CommandIDs.gitOpenGitignore,
+      CommandIDs.gitShowDiff,
+      CommandIDs.gitInit,
+      CommandIDs.gitClone,
```
Comment (fcollonval) [inline src/index.ts:204]: You must not add this one here as it is defined in another plugin.

```suggestion
```

Instead you should add the same logic in the clone plugin there:

https://github.com/jupyterlab/jupyterlab-git/blob/dda8e82b9b2751e1b2006f71cb101accef261d10/src/cloneCommand.ts#L20

**Review by fcollonval — CHANGES_REQUESTED** [review]
Thanks @tsabbir96 

Don't worry about the integration tests failing. They are not related to your work.

I only have small comments and this will be ready to go.

File: src/index.ts (hunk 3 above)
Comment (tsabbir96) [reply src/index.ts:204]: > You must not add this one here as it is defined in another plugin.
> 
> Instead you should add the same logic in the clone plugin there:
> 
> https://github.com/jupyterlab/jupyterlab-git/blob/dda8e82b9b2751e1b2006f71cb101accef261d10/src/cloneCommand.ts#L20

@fcollonval Hi, thanks for the feedback. Could you kindly elaborate a bit for me on what you meant by "Instead you should add the same logic in the clone plugin there". Thank you once again.

Comment (fcollonval) [reply src/index.ts:204]: `gitCloneCommandPlugin` is an extension plugin as the `plugin` you modified there:

https://github.com/jupyterlab/jupyterlab-git/blob/fba3e2fe1def7d9f051221cf2d7e2aac26652be5/src/index.ts#L44

So you should also add the `ICommandPalette` token as requirement and then call,

```ts
palette.addItem({ command: CommandIDs.gitClone, category: trans.__('Git Operations') })
```

**Review by fcollonval — APPROVED** [review]

**Pull request merged**